### PR TITLE
core: lb: Removed has_force_density logic.

### LINKED ISSUE #2078: lb_cpu without EXTERNAL_FORCES silently ignores a set ext_force_density
```python
import espressomd
import espressomd.lb

system = espressomd.System(box_l=[10.0]*3, time_step=1.0)
system.cell_system.skin = 0.01

lbf = espressomd.lb.LBFluid(                                  
		    visc=1.0,                                                           
		    dens=1.0,                                                           
		    agrid=1.0,                                                          
		    tau=1.0,                                                          
		    fric=1.0,
		    ext_force_density=[.1,.2,.3])                                    

system.thermostat.set_lb(kT=0.)
system.actors.add(lbf)
system.integrator.run(steps=100)
print system.analysis.analyze_linear_momentum(), lbf[5,5,5].velocity
system.actors.remove(lbf)
print("After removal")
print system.analysis.analyze_linear_momentum()
system.actors.add(lbf)
print("After re-adding")
print system.analysis.analyze_linear_momentum()
system.integrator.run(1)
print system.analysis.analyze_linear_momentum()
```

### PR DESCRIPTION
**Pull request by fweik**
Fixes #2078.

Description of changes:
 - Removed `has_force_density` logic, and
   always apply forces. The kernel was executed anyway.
 - Made LB independent of `EXTERNAL_FORCES`.

### CODE REVIEW

File: src/core/lb.cpp [hunk 1]
```diff
@@ -2468,21 +2434,28 @@ inline void lb_apply_forces(Lattice::index_t index, double *mode) {
...
+  lbfields[index].force_density[1] = lbpar.ext_force_density[1] * lbpar.agrid *
+                                     lbpar.agrid * lbpar.tau * lbpar.tau;
+  lbfields[index].force_density[2] = lbpar.ext_force_density[2] * lbpar.agrid *
+                                     lbpar.agrid * lbpar.tau * lbpar.tau;
+}
+
+/** Resets the forces on the fluid nodes */
+void lb_reinit_force_densities() {
+  for (Lattice::index_t index = 0; index < lblattice.halo_grid_volume;
+       index++) {
```
Comment (KaiSzuttor) [inline src/core/lb.cpp:2450]: preincrement

File: src/core/lb.cpp [hunk 2]
```diff
@@ -2983,7 +2945,7 @@ void lb_lbfluid_get_interpolated_velocity(const Vector3d &p, double *v) {
  * halo regions before treating the ghost particles. Moreover, after
  * determining the ghost couplings, we have to communicate back the
  * halo region such that all local lattice nodes have the correct values.
- * Thus two communication phases are involved which will most likely be
+ * Thus two communication phass are involved which will most likely be
```
Comment (KaiSzuttor) [inline src/core/lb.cpp:2948]: improved for the worse?

**Review by KaiSzuttor — APPROVED** [review]
LGTM bes. comments

File: src/core/lb.cpp (hunk 1 above)
Comment (fweik) [reply src/core/lb.cpp:2450]: that's an int

**Pull request merged**